src/alr/alr-commands-init.adb

```diff
@@ -18,38 +19,144 @@ package body Alr.Commands.Init is
    --------------
 
    procedure Generate (Cmd : Command) is
-      Name  : constant String := Argument (1);
+      use Alr.Os_Lib;
+
+      package TIO renames Ada.Text_IO;
+
+      For_Library : constant Boolean := not Cmd.Bin;
+      Name        : constant String := Argument (1);
+      Lower_Name  : constant String := Utils.To_Lower_Case (Name);
+      Mixed_Name  : constant String := Utils.To_Mixed_Case (Name);
+
+      Directory     : constant String :=
+        (if Cmd.In_Place
+         then Alire.Directories.Current
+         else Alire.Directories.Current / Name);
+      Src_Directory : constant String := Directory / "src";
+
+      File : TIO.File_Type;
+
+      procedure Put_New_Line;
+      procedure Put_Line (S : String);
+      --  Shortcuts to write to File
+
+      procedure Generate_Project_File;
+      --  Generate a project file for this crate
+
+      procedure Generate_Root_Package;
+      --  Generate the specification for the root package of this crate
+
+      procedure Generate_Program_Main;
+      --  Generate the procedure body for the program main of this crate
+
+      ---------------------------
+      -- Generate_Project_File --
+      ---------------------------
+
+      procedure Generate_Project_File is
+         Filename : constant String := Directory / (Lower_Name & ".gpr");
+      begin
+         TIO.Create (File, TIO.Out_File, Filename);
+         Put_Line ("project " & Mixed_Name & " is");
+         Put_New_Line;
+         if For_Library then
+            Put_Line ("   for Library_Name use """ & Mixed_Name & """;");
+            Put_Line ("   for Library_Version use ""0.0.0"";");
+            Put_New_Line;
+         end if;
+         Put_Line ("  for Source_Dirs use (""src"");");
+         Put_Line ("  for Object_Dir use ""obj"";");
+         if For_Library then
+            Put_Line ("  for Library_Dir use ""lib"";");
+         else
+            Put_Line ("  for Exec_Dir use ""bin"";");
+            Put_Line ("  for Main use (""" & Lower_Name & ".adb"");");
+         end if;
+         Put_New_Line;
+         Put_Line ("   package Builder is");
+         Put_Line ("      for Switches (""ada"") use (""-j0"", ""-g"");");
+         Put_Line ("   end Builder;");
+         Put_New_Line;
+         Put_Line ("   package Compiler is");
+         Put_Line ("      for Switches (""ada"") use");
+         Put_Line ("        (""-gnatVa"", ""-gnatwa"", ""-g"", ""-O2"",");
+         Put_Line ("         ""-gnata"", ""-gnato"", ""-fstack-check"");");
+         Put_Line ("   end Compiler;");
+         Put_New_Line;
+         Put_Line ("   package Binder is");
+         Put_Line ("      for Switches (""ada"") use (""-Es"");");
+         Put_Line ("   end Binder;");
+         Put_New_Line;
+         Put_Line ("end project " & Mixed_Name & ";");
```

Comment:
"end project " --> "end "

Reply:
Oops, I tested the init command but forgot to do a build after it… will fix, thanks!